src/main/groovy/io/seqera/wave/service/persistence/impl/SurrealPersistenceService.groovy

```diff
@@ -257,8 +261,23 @@ class SurrealPersistenceService implements PersistenceService {
         final vulnerabilities = scanRecord.vulnerabilities ?: List.<ScanVulnerability>of()
 
         // save all vulnerabilities
+        int count = 0
         for( ScanVulnerability it : vulnerabilities ) {
-            surrealDb.insertScanVulnerability(authorization, it)
+            if ( ++count > scanConfig.vulnerabilityLimit )
```

Comment:
Fear too simple. it should be filtered by vulnerability levels ie. first the critical ones, then high, etc until reach the 50 cut off. 

It may be better doing it adding a `filter` method in this class 

https://github.com/seqeralabs/wave/blob/d1d1a6228bac8013632742bfa6654438f1f9e1d9/src/main/groovy/io/seqera/wave/service/scan/TrivyResultProcessor.groovy#L35-L37

Reply:
I thought vulnerabilities are already sorted, I will check and add a filter

Reply:
I see, I was not aware of that, still better to keep the filtering in its own method